pkg/clientaccess/token.go

```diff
@@ -288,6 +299,32 @@ func get(u string, client *http.Client, username, password string) ([]byte, erro
 	return ioutil.ReadAll(resp.Body)
 }
 
+// put makes a request to a url using a provided client, username, and password
+// only an error is returned
+func put(u string, body []byte, client *http.Client, username, password string) error {
+	req, err := http.NewRequest(http.MethodPut, u, bytes.NewBuffer(body))
+	if err != nil {
+		return err
+	}
+
+	if username != "" {
+		req.SetBasicAuth(username, password)
+	}
+
+	resp, err := client.Do(req)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	respBody, _ := ioutil.ReadAll(resp.Body)
+	if resp.StatusCode != http.StatusOK {
+		return fmt.Errorf("%s: %s %s", u, resp.Status, string(respBody))
```

Comment:
Don't include the body in the error; since we're making an authenticated request it could have something confidential in it that we wouldn't want logged.
```suggestion
		return fmt.Errorf("%s: %s", u, resp.Status)
```

Reply:
The respBody is the error message that came out of the server. I want that error to be reported to the user. Is there another way I should be reporting it?

Reply:
Generally for proper REST stuff there are typed errors with a human-readable `message` field meant for displaying to users; if the error body can be decoded to the expected error type then it should be safe to put in an error that might be logged; anything else is just a generic error that you'd need to enable debugging to log since it may contain a stack trace or something else confidential.

That's probably overkill for what we're doing here though, so the heck with it.

Reply:
Reworked error handling. Now the server generates a unique integer and logs that + the error. The CLI simply reports that integer to the user and are promted to go look at the server logs.